samples/web/content/apprtc/apprtc.py

```diff
@@ -495,12 +515,109 @@ def write_response(self, result, params, messages):
       'result': result,
       'params': params
     }))
+    
+  def report_error(self, error):
+    self.write_response(error, {}, [])
 
   def write_room_parameters(self, room_id, client_id, messages, is_initiator):
     params = get_room_parameters(self.request, room_id, client_id, is_initiator)
     self.write_response('SUCCESS', params, messages)
+    
+  def handle_call(self, msg, room_id):
+    """Handles a new call request."""
+    if not util.has_msg_fields(msg, ((PARAM_CALLER_GCM_ID, basestring),
+        (PARAM_CALLEE_ID, basestring))):
+      self.report_error(constants.RESPONSE_INVALID_ARGUMENT)
+      return
+      
+    callerId = msg[PARAM_CALLER_GCM_ID]
+    calleeId = msg[PARAM_CALLEE_ID]
+    # Look up and validate caller by gcm id.
+    callerRecords = gcmrecord.GCMRecord.get_by_gcm_id(callerId, True)
```

Comment:
We are not planning to implement sending verification code from the client in the first milestone, so #2 is not appropriate.
For #1, it will leave 'incorrect' data in the datastore, which is less preferable than a simple code change in the future.